crates/kitsune_p2p/kitsune_p2p/src/spawn/actor/discover.rs

```diff
@@ -112,7 +112,7 @@ pub(crate) fn peer_connect(
         .url_list
         .get(0)
         .cloned()
-        .ok_or_else(|| KitsuneP2pError::from("no url"));
+        .ok_or_else(|| KitsuneP2pError::from("no url - agent is likely offline"));
```

Comment:
I think this is fine but I wonder who is handling this error as it looks like below you set the urls to empty?

Reply:
I just had to move the question mark down into the async block. that `let url = url?;` will cause the error to propagate up to whomever is calling it... i.e. the gossip loop or the rpc_single, etc.